homeassistant/components/intellifire/fan.py

```diff
@@ -0,0 +1,171 @@
+"""Fan definition for Intellifire."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+import math
+from typing import Any
+
+from intellifire4py import IntellifireControlAsync, IntellifirePollData
+
+from homeassistant.components.fan import (
+    FanEntity,
+    FanEntityDescription,
+    FanEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.percentage import (
+    ordered_list_item_to_percentage,
+    percentage_to_ordered_list_item,
+)
+
+from .const import DOMAIN, LOGGER
+from .coordinator import IntellifireDataUpdateCoordinator
+from .entity import IntellifireEntity
+
+
+@dataclass
+class IntellifireFanRequiredKeysMixin:
+    """REquired keys for fan entity."""
+
+    set_fn: Callable[[IntellifireControlAsync, int], Awaitable]
+    value_fn: Callable[[IntellifirePollData], bool]
+    data_field: str
+    named_speeds: list[str]
+
+
+@dataclass
+class IntellifireFanEntityDescription(
+    FanEntityDescription, IntellifireFanRequiredKeysMixin
+):
+    """Describes a fan entity."""
+
+
+INTELLIFIRE_FANS: tuple[IntellifireFanEntityDescription, ...] = (
+    IntellifireFanEntityDescription(
+        key="fan",
+        name="Fan",
+        set_fn=lambda control_api, speed: control_api.set_fan_speed(
+            fireplace=control_api.default_fireplace, speed=speed
+        ),
+        value_fn=lambda data: data.fanspeed,
+        data_field="fanspeed",
+        named_speeds=[
+            "quiet",
```

Comment:
Is there a standard on how these should be presented?

* `quiet`
* `Quiet`
* `QUIET`